### PR TITLE
V1C-205 – fix(form): show a loader when form is being submitted

### PR DESCRIPTION
**Pull request by mohsinht**
A form activity completion is dependent on 2 mutations:

1. Evaluate Form Rules
2. Submit Form Response

Only on success of both of these mutations, we mark the activity as completed.

The blinking of last question is because of the delay between `EvaluateFormRules` and `SubmitFormResponse`. See this example:

The last question was shown just after completion of `EvaluateFormRules` mutation.

----

After adding another loader for 'Submitting Form' process, here's the result:

Instead of showing the last question again, we showed the 'Submitting form...' loader text

### CODE REVIEW

File: src/hooks/useSubmitForm/useSubmitForm.ts [hunk 1]
```diff
@@ -51,5 +52,6 @@ export const useSubmitForm = (activity: Activity): UseFormActivityHook => {
   return {
     onSubmit,
     disabled: isSubmitting,
+    isSubmitting,
```
Comment (danijel) [inline src/hooks/useSubmitForm/useSubmitForm.ts:55]: we already return this in a line above as disabled prop.

**Comment by danijel** [comment]
are you sure this happens because of polling, did you verify that with polling disabled?

File: src/hooks/useSubmitForm/useSubmitForm.ts (hunk 1 above)
Comment (mohsinht) [reply src/hooks/useSubmitForm/useSubmitForm.ts:55]: yes, but I assumed it to be internal logic of the process to decide when should the submitting be disabled.

I'll change it 👍

**Comment by mohsinht** [comment]
> are you sure this happens because of polling, did you verify that with polling disabled?

Not entirely because of polling. a form activity completion is dependent on 2 mutations:

1. Evaluate Form Rules
2. Submit Form Response

Only on success of both of these mutations, we mark the activity as completed.

The blinking of last question is because of the delay between `EvaluateFormRules` and `SubmitFormResponse`. See this example:


The last question was shown just after completion of `EvaluateFormRules` mutation.

**Review by FlavioF — APPROVED** [review]

**Pull request merged**